exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserSession.java

```diff
@@ -116,14 +137,38 @@ public OptionManager getOptions() {
     return sessionOptions;
   }
 
-  public DrillUser getUser() {
-    return user;
-  }
-
   public UserCredentials getCredentials() {
     return credentials;
   }
 
+  /**
+   * Replace current user credentials with the given user's credentials, if authorized.
+   *
+   * @param delegatorName delegator name
+   * @throws DrillRuntimeException if credentials cannot be replaced
+   */
+  public void replaceUserCredentials(final String delegatorName) {
+    assert enableDelegation;
```

Comment:
No need for assert, preconditions makes sure we get Exception instead of typically uncaptured Error subclass and this isn't perf sensitive.